src/data-workspace/data-value-mutations/optimistic-update-helpers.js

```diff
@@ -0,0 +1,97 @@
+export function findDataValueIndex(
+    previousDataValues,
+    dataValueMutationParams
+) {
+    const { co, de, ou, pe } = dataValueMutationParams
+
+    return previousDataValues.findIndex(
+        (dataValue) =>
+            dataValue.categoryOptionCombo === co &&
+            dataValue.dataElement === de &&
+            // not sure the following two matchers are necessary
```

Comment:
I don't see why matching on the orgUnit and period would be necessary given that the previous data values should only be relevant to the given dataset/period/orgunit/attribute options. I guess I would vote to remove these extra criteria or add in the ones for attribute options